### PR TITLE
[IMPORTANT] Fixed documentElement.innerHTML setter children popup.

### PR DESCRIPTION
**Pull request by Indomian**
Fixed situation when documentElement.innerHTML is set with full HTML
document. In such situation parser just creates new documentElement
under documentElement, which is wrong. Older version of popup didn't
notice text nodes before <html> thats why using _childNodeFind on document
could not find body and created new one.
    
    Signed-off-by: Egor Bolgov <e.bolgov@semrush.com>

### CODE REVIEW

File: tests/src/HTMLElement/elements/HTMLDocumentElement.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*global test*/
+import assert from 'proclaim';
```
Comment (christophehurpeau) [inline tests/src/HTMLElement/elements/HTMLDocumentElement.js]: you can do: `import { strictEqual as expect, throws } from 'proclaim'`

Comment (Indomian) [inline tests/src/HTMLElement/elements/HTMLDocumentElement.js]: Copy/paste from other test :) I'll update - problems with eslint on CI

File: tests/src/HTMLElement/elements/HTMLDocumentElement.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const expect = assert.strictEqual;
+const throws = assert.throws;
+
+const lib = '../../../../lib/';
+
+const Document = require(lib + 'Document');
+const Doctype = require(lib + 'Doctype');
+
+const fileText = readFileSync(__dirname + '/../../../data/linkstest.html', { encoding: 'utf-8' });
```
Comment (christophehurpeau) [inline tests/src/HTMLElement/elements/HTMLDocumentElement.js]: I don't think you need this big html to create the test below, right ? What you want to test, if I got it right, is:

- `document.documentElement.innerHTML = '<html></html>';`
- `document.documentElement.innerHTML = '<html><body>test</body></html>';`
- `document.documentElement.innerHTML = '<html><head><title>test</title></head></html>';`
- `document.documentElement.innerHTML = '<html><head><title>test</title></head><body>test</body></html>';`

And copy.

What do you think ?

Comment (christophehurpeau) [inline tests/src/HTMLElement/elements/HTMLDocumentElement.js]: And:
- `document.documentElement.innerHTML = '<body>test</body>';`
of course :)

Comment (Indomian) [inline tests/src/HTMLElement/elements/HTMLDocumentElement.js]: Well, you are right. I used that document, which were failing for me. I just moved tests from my application to html-document to reproduce problem, so I can find a problem. After I just left it as is.
I have some more plans on this document later (especially for easier testing of selectors etc.).
I can update tests if you think they are not clear now.

Comment (Indomian) [inline tests/src/HTMLElement/elements/HTMLDocumentElement.js]: Ok, I got the point. I will update tests in 10 minutes.

Comment (christophehurpeau) [inline tests/src/HTMLElement/elements/HTMLDocumentElement.js]: Thanks !

File: tests/src/HTMLElement/elements/HTMLDocumentElement.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    strictEqual(document.body.innerHTML, 'test');
+});
+
+test('should do same as above on copy', () => {
+    let document = new Document();
+    document.documentElement.innerHTML = '<html><head></head><body>test</body></html>';
+    let documentClone = document.documentElement.cloneNode(true);
+    strictEqual(documentClone.children.length, 2);
+    strictEqual(documentClone.getElementsByTagName('body')[0].innerHTML, 'test');
+});
```
Comment (christophehurpeau) [inline tests/src/HTMLElement/elements/HTMLDocumentElement.js]: can you add a test without the `<html>` too ? thanks :)

**Pull request merged**